update default version of API to v1 instead of v1beta3 [#101]

jonmoter opened the pull request:
Updates defaults in code, as well as tests.

See issue #101.

abonas commented:
@jonmoter Hi, and thanks a lot for your contribution.
Couple of comments from briefly looking at this:
all test json files names remained with "b3" name which stands for beta 3.
a. it makes sense to rename them as part of this change.
b. is the format kept 100% identical in k8s between beta3 to v1? so you can simply change the api version in the json and that would be enough? (no other fields added, fields renamed or moved between sections, etc.?)
c. in order to link properly the referenced github issue, please make sure to follow this convention in your commit comment:
https://help.github.com/articles/closing-issues-via-commit-messages/

abonas commented:
cc @simon3z

jonmoter commented:
Okay, I've renamed the files to remove the "_b3" in the filename.

I went over the Kubernetes docs describing [changes from v1beta3 to v1](https://github.com/kubernetes/kubernetes/blob/master/docs/api.md#v1-conversion-tips-from-v1beta3), and found a few small changes that were required. I updated the tests to reflect that.

Let me know if you have any other feedback.

abonas commented:
> c. in order to link properly the referenced github issue, please make sure to follow this convention in your commit comment:
> https://help.github.com/articles/closing-issues-via-commit-messages/

@jonmoter LGTM except for not linking commit msg

jonmoter commented:
Would you like me to rebase with the commit message fixed?

> On Sep 14, 2015, at 4:51 PM, abonas <notifications@github.com> wrote:
> 
> c. in order to link properly the referenced github issue, please make sure to follow this convention in your commit comment:
> https://help.github.com/articles/closing-issues-via-commit-messages/
> 
> @jonmoter LGTM except for not linking commit msg
> 
> —
> Reply to this email directly or view it on GitHub.
>

abonas commented:
You don't need to rebase for that, just fix the commit msg, then repush.
You can also squash the 3 commits, they all handle the same topic
On Sep 15, 2015 12:53 AM, "Jon Moter" <notifications@github.com> wrote:

> Would you like me to rebase with the commit message fixed?
>
> > On Sep 14, 2015, at 4:51 PM, abonas <notifications@github.com> wrote:
> >
> > c. in order to link properly the referenced github issue, please make
> sure to follow this convention in your commit comment:
> > https://help.github.com/articles/closing-issues-via-commit-messages/
> >
> > @jonmoter LGTM except for not linking commit msg
> >
> > —
> > Reply to this email directly or view it on GitHub.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/abonas/kubeclient/pull/120#issuecomment-140215932>.
>

jonmoter commented:
@abonas okay, i've squashed the commits and fixed the commit message.

abonas commented on test/test_node.rb:
```diff
@@ -4,10 +4,10 @@
 class TestNode < MiniTest::Test
   def test_get_from_json_v3
```
missed a rename here in the method name

abonas commented on test/test_pod.rb:
```diff
@@ -4,10 +4,10 @@
 class TestPod < MiniTest::Test
   def test_get_from_json_v3
```
missed a rename here in the method name

abonas commented on test/test_replication_controller.rb:
```diff
@@ -4,10 +4,10 @@
 class TestReplicationController < MiniTest::Test
   def test_get_from_json_v3
```
missed a rename here in the method name

abonas commented:
@jonmoter great. I just noticed couple of method still refer to v3, please remove that too and it's good to be merged.

jonmoter commented:
D'oh, good catch.  I'll fix and squash.

jonmoter commented:
@abonas fixed.  Should be good to go.

abonas commented:
:+1:
